### ISSUE TITLE
Ability to weakly reference an instance of a class lost due to mutliple inheritance

### ISSUE THREAD

**Issue by markshannon** [python/cpython#95710]
This is related to https://github.com/python/cpython/issues/95589 due to underlying mechanism, but the behavior is distinct.

```Python 
class C: 
    pass

class D(int, C):
    pass
>>> c = C()
>>> weakref.ref(c)
<weakref at 0x7fa5b2af9c60; to 'C' at 0x7fa5b2e501c0>
>>> d = D()
>>> weakref.ref(d)
TypeError: cannot create weak reference to 'D' object
>>> isinstance(d, C)
True
```
If I can weakly reference an instance of `C`, then I should be able to weakly reference an instance of `D`. Instances of `D` are instances of `C`

**Comment by Mandera** [comment]
I don't think this has much to do with inheritance. It's caused by `int` not supporting weakref.

See https://stackoverflow.com/a/52011601/3936044

`int` has a `__weakrefoffset__` of 0 meaning it has no weakref pointer, remove the `int` baseclass and it works fine
